tests/integration/clientv3/user_test.go

```diff
@@ -55,6 +55,66 @@ func TestUserError(t *testing.T) {
 	}
 }
 
+func TestAddUserAfterDelete(t *testing.T) {
```

Comment:
Have you confirmed that the test case can reproduce the issue?

Reply:
Confirmed the test case if the [defensive commit](https://github.com/etcd-io/etcd/blob/f66b95d3b1374beb56dc772c587832f15bb6e697/server/storage/schema/auth.go#L114) is removed. Because auth force commits for every change, this is not actually a problem for auth right now. 

I am adding this test for completeness.

Reply:
Wow, nice find. This defensive commit seems scary, similar to https://github.com/etcd-io/etcd/pull/17119 we noticed some inconsistency and didn't root cause it.